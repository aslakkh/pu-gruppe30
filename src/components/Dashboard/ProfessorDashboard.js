import React, { Component } from 'react'
import {Tabs, Tab} from 'react-bootstrap'
import ManageCourses from './ManageCourses'
import CoursesList from './CoursesList'

import firebase from 'firebase';

export default class ProfessorDashboard extends Component {
/*
    TAB-view: "my courses" and "manage all courses"
    allow professor to add and delete courses in manage-courses, as well as add courses to his/her courseslist
    TODO:
    - add courses to professor and to courses
    - let professor choose a course already in database to be assigned to
*/



    componentDidMount(){
    }

    




    render () {
        return (
            <div>
                <h4>Professor dashboard working</h4>
                <Tabs defaultActiveKey={0} id="dashboardTab">
                    <Tab eventKey={0} title="My Courses"><CoursesList courses={this.props.courses}/></Tab>
                    <Tab eventKey={1} title="Manage All Courses"><ManageCourses/></Tab>
                </Tabs>
            </div>
        )
    }
}