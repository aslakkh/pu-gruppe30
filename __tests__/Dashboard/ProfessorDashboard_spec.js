import React from 'react';
import { shallow } from 'enzyme';
import TestUtils from "react-addons-test-utils";
import ProfessorDashboard from '../../src/components/Dashboard/ProfessorDashboard.js';

describe("ProfessorDashboard", function(){
    var renderer, component;
    renderer = TestUtils.createRenderer();
    renderer.render(
        <ProfessorDashboard />
    );
    component = renderer.getRenderOutput();

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    

})

