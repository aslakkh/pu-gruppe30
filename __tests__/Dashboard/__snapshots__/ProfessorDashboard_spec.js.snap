// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfessorDashboard matches snapshot 1`] = `
<Uncontrolled(Tabs)
  defaultActiveKey={0}
  id="dashboardTab"
>
  <Tab
    eventKey={0}
    title="My Courses"
  >
    <CoursesList
      courses={undefined}
    />
  </Tab>
  <Tab
    eventKey={1}
    title="Manage All Courses"
  >
    <ManageCourses />
  </Tab>
</Uncontrolled(Tabs)>
`;
